### PR TITLE
[sensors_plus] Actually cleanup in onDetachedFromEngine

### LINKED ISSUE #752: [sensors_plus] Actually cleanup in onDetachedFromEngine
## System info

Platform the Issue occurs on: Android
Plugin name: sensors_plus
Plugin version: 1.2.2

## Steps to Reproduce

1. Collect data
2. Kill app from recents
3. You'll get a ton of these: `W[/FlutterJNI]()(13631): Tried to send a platform message to Flutter, but FlutterJNI was detached from native C++. Could not send.`

### PR DESCRIPTION
**Pull request by Rexios80**
## Description

Plugin was not cleaning up in onDetachedFromEngine on Android

Not sure how to test this or if it's even testable

Closes #752

## Related Issues

#752

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated the version in `pubspec.yaml` and `CHANGELOG.md`.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/fluttercommunity/plus_plugins/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

**Comment by vbuberen** [comment]
Thanks for updating your PR. I will get back to the issue and this fix in a few days when have more free time.

**Comment by Rexios80** [comment]
Any updates on this?

File: packages/sensors_plus/sensors_plus/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.3.1
+
+- Cleanup resources on Android in onDetachedFromEngine
```
Comment (vbuberen) [inline packages/sensors_plus/sensors_plus/CHANGELOG.md:3]:
```suggestion
- Fix: deregister listeners on Android in `onDetachFromEngine` to not receive sensors events after app was killed
```

**Review by vbuberen — APPROVED** [review]
Looks good to me. Thanks for your contribution.
I have just updated the changelog to be more explicit on what changed.

P.S. It is hard to find free time these days after war started in my country. Thus, the delay.

**Comment by clouddevil** [comment]
After update to 1.3.1 I get this **issue**:
```
D/AndroidRuntime( 8367): Shutting down VM
E/AndroidRuntime( 8367): FATAL EXCEPTION: main
E/AndroidRuntime( 8367): Process: com.example.sensor_test, PID: 8367
E/AndroidRuntime( 8367): java.lang.RuntimeException: Unable to destroy activity {com.example.sensor_test/com.example.sensor_test.MainActivity}: kotlin.UninitializedPropertyAccessException: lateinit property sensorEventListener has not been initialized
....
I/Process ( 4594): Sending signal. PID: 4594 SIG: 9
```

**Step to reproduce:**
- create new flutter project
- add dependency sensors_plus: ^1.3.1
- flutter run on Android (i run at api 29)
- press back button to exit app
- check log messages

<details>
<summary>flutter doctor</summary>

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 2.10.4, on macOS 12.3.1 21E258 darwin-arm, locale en-RU)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 13.3.1)
[✓] Android Studio (version 2021.1)
[✓] Android Studio (version 2020.3)
[✓] VS Code (version 1.66.1)
[✓] Connected device (3 available)
[✓] HTTP Host Availability
```
</details>

**Comment by Rexios80** [comment]
Gosh darnit late init properties be like. I'll fix it.

**Comment by Rexios80** [comment]
Wait a second. This should have already been a problem. I don't think this is related to this change.

**Pull request merged**